python/ql/src/semmle/python/objects/ObjectAPI.qll

```diff
@@ -514,6 +514,13 @@ abstract class FunctionValue extends CallableValue {
     predicate isOverriddenMethod() {
         exists(Value f | f.overrides(this))
     }
+
+    /** Gets a class that this function may return */
+    ClassValue getAnInferredReturnType() {
+        result = this.(BuiltinFunctionValue).getAReturnType()
+        or
+        result = this.(BuiltinMethodValue).getAReturnType()
```

Comment:
I think it might be better to push the implementations of `getAnInferredReturnType` and `getAReturnType` into `CallableObjectInternal` and its subclasses, and then just call this from this predicate.